app/indexers/concerns/indexes_citation_metadata.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module IndexesCitationMetadata
+  extend ActiveSupport::Concern
+
+  def generate_solr_document
+    super.tap do |doc|
+      next doc if object.bibliographic_citation.empty?
+
+      citation = AnyStyle.parse(object.bibliographic_citation.first)
+
+      next doc if citation.blank?
+
+      entry = citation.first
+
+      doc['citation_journal_title_ss'] = entry[:"container-title"].blank? ? "" : entry[:"container-title"].first
+      doc['citation_volume_ss'] = entry[:"volume"].blank? ? "" : entry[:"volume"].first
+      doc['citation_issue_ss'] = entry[:"issue"].blank? ? "" : entry[:"issue"].first
+      doc['citation_firstpage_ss'] = entry[:"pages"].blank? ? "" : entry[:"pages"].first.split('–', 2).first
+      doc['citation_lastpage_ss'] = entry[:"pages"].blank? ? "" : entry[:"pages"].first.split('–', 2).last

```

Comment:
something that could help cut complexity down is using Ruby's [safe navigation operator](https://ruby-doc.org/2.7.7/syntax/calling_methods_rdoc.html#label-Safe+navigation+operator) to return the first value or nil. 

ex: `doc['citation_volume_ss'] = entry[:volume]&.first`